### PR TITLE
CI move to ALPS (daint-mc -> eiger)

### PR DESCRIPTION
**Pull request by rasolca**
Depends on #1197

### CODE REVIEW

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

File: ci/common-ci.yml [hunk 1]
```diff
@@ -57,51 +54,89 @@ stages:
...
+    RUNNER: ".container-runner-eiger"
+    SLURM_CONSTRAINT: mc
+    THREADS_MAX_PER_TASK: 32
+    THREADS_PER_NODE: 256
+
+.build_for_alps_gh200:
+  variables:
+    RUNNER: ".container-runner-todi"
+    SLURM_CONSTRAINT: gpu
+    SLURM_RESERVATION: "debug"
```
Comment (msimberg) [inline ci/common-ci.yml:132]:
```suggestion
    SLURM_ACCOUNT: "g154"
    SLURM_RESERVATION: "daint"
```
?

File: ci/common-ci.yml [hunk 2]
```diff
@@ -57,51 +54,89 @@ stages:
...
+  variables:
+    COMMON_SPACK_ENVIRONMENT: ci/docker/common.yaml
+
+.build_deps_common_gh200:
+  extends:
+    - .container-builder-cscs-gh200
+    - .build_deps_common_base
+  variables:
+    COMMON_SPACK_ENVIRONMENT: ci/docker/common-gh200.yaml
+    SLURM_RESERVATION: "daint"
```
Comment (msimberg) [inline ci/common-ci.yml:72]:
```suggestion
    SLURM_ACCOUNT: "g154"
    SLURM_RESERVATION: "daint"
```

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
clang15 C++20 problematic?
- problem with MPI https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/jobs/8521984909
- problem with eigensolver https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/jobs/8522697491

@msimberg @biddisco any idea?

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Pull request merged**